pkg/virtual/apiexport/builder/build.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/kcp-dev/logicalcluster"
+
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	kcpinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/virtual/apiexport/controllers/apireconciler"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework"
+	virtualdynamic "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apidefinition"
+	"github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/apiserver"
+	dynamiccontext "github.com/kcp-dev/kcp/pkg/virtual/framework/dynamic/context"
+)
+
+const VirtualWorkspaceName string = "apiexport"
+
+func BuildVirtualWorkspace(
+	rootPathPrefix string,
+	dynamicClusterClient dynamic.ClusterInterface,
+	kcpClusterClient kcpclient.ClusterInterface,
+	wildcardKcpInformers kcpinformer.SharedInformerFactory,
+) framework.VirtualWorkspace {
+	if !strings.HasSuffix(rootPathPrefix, "/") {
+		rootPathPrefix += "/"
+	}
+
+	readyCh := make(chan struct{})
+
+	return &virtualdynamic.DynamicVirtualWorkspace{
+		Name: VirtualWorkspaceName,
+
+		RootPathResolver: func(path string, requestContext context.Context) (accepted bool, prefixToStrip string, completedContext context.Context) {
+			completedContext = requestContext
+
+			if !strings.HasPrefix(path, rootPathPrefix) {
+				return
+			}
+
+			// Incoming requests to this virtual workspace will look like:
+			//  /services/apiexport/root:org:ws/<apiexport-name>/clusters/*/api/v1/configmaps
```

Comment:
This is missing the apiExport cluster name in the prefix, right?

Reply:
that's `root:org:ws`, no?